subprojects/workers/src/main/java/org/gradle/workers/internal/DefaultProcessWorkerSpec.java

```diff
@@ -16,23 +16,56 @@
 
 package org.gradle.workers.internal;
 
-import com.google.common.collect.Maps;
+import com.google.common.collect.ImmutableMap;
 import org.gradle.api.Action;
 import org.gradle.api.model.ObjectFactory;
+import org.gradle.internal.os.OperatingSystem;
 import org.gradle.process.JavaForkOptions;
 import org.gradle.workers.ClassLoaderWorkerSpec;
 import org.gradle.workers.ProcessWorkerSpec;
 
 import javax.inject.Inject;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.regex.Pattern;
 
 public class DefaultProcessWorkerSpec extends DefaultClassLoaderWorkerSpec implements ProcessWorkerSpec, ClassLoaderWorkerSpec {
+    /**
+     * Environment variables inherited automatically on Unix systems.
+     *
+     * See <a href="https://www.gnu.org/software/gettext/manual/html_node/Locale-Environment-Variables.html">Locale Environment Variables for gettext</a>
+     */
+    private static final Pattern INHERITED_UNIX_ENVIRONMENT = Pattern.compile("(LANG|LANGUAGE|LC_.*)");
+
     protected final JavaForkOptions forkOptions;
 
     @Inject
     public DefaultProcessWorkerSpec(JavaForkOptions forkOptions, ObjectFactory objectFactory) {
         super(objectFactory);
         this.forkOptions = forkOptions;
-        this.forkOptions.setEnvironment(Maps.newHashMap());
+        this.forkOptions.setEnvironment(sanitizeEnvironment(forkOptions));
+    }
+
+    /**
+     * Inherit as little as possible from the parent process' environment.
+     *
+     * On Unix systems we need to pass a few environment variables to make sure
+     * the file system is accessed with the right encoding.
+     */
+    private static Map<String, Object> sanitizeEnvironment(JavaForkOptions forkOptions) {
+        if (!OperatingSystem.current().isUnix()) {
+            return ImmutableMap.of();
+        }
+        Map<String, Object> environment = forkOptions.getEnvironment();
```

Comment:
Is it possible that `forkOptions.getEnvironment()` return immutable map?

Reply:
If we think this is meaningful, we can try to improve things in a separate PR.

This PR might be backported as far as 8.1, so I intentionally kept it as small as I could.

Reply:
But if user passes immutable map to forOptions, then this will blow up, no?

Reply:
Ah, I see what you mean. I thought you meant we should make it immutable. I think you're right, let me fix this.

Reply:
Actually, no, let's leave it as is. This code was already calling `forkOptions.setEnvironment()` in the constructor, which would have blown up with an `ImmutableJavaForkOptions`. So we aren't making things worse here.

This should be improved, but let's do that in a separate PR.

Reply:
I don't like the current semantic of that method: it returns a `Map` but potentially by modifying the original `Map`. Which means in some cases the assignment is not needed. That feels like a code smell to me. I would rather have a clear new `Map` being built always.

Reply:
Alright then. Should be fixed now.